rasa/core/policies/rule_policy.py

```diff
@@ -828,15 +839,27 @@ def predict_action_probabilities(
         loop_happy_path_action_name = self._find_action_from_loop_happy_path(tracker)
         if loop_happy_path_action_name:
             self._prediction_source = LOOP_RULES
-            return self._prediction_result(loop_happy_path_action_name, tracker, domain)
+            return PolicyPrediction(
+                self._prediction_result(loop_happy_path_action_name, tracker, domain),
+                policy_priority=self.priority,
+            )
 
-        rules_action_name, source = self._find_action_from_rules(tracker, domain)
+        rules_action_name, source, entered_unhappy_path = self._find_action_from_rules(
+            tracker, domain
+        )
         # we want to remember the source even if rules didn't predict any action
         self._prediction_source = source
+
+        policy_events = []
+        if entered_unhappy_path:
+            policy_events.append(LoopInterrupted(True))
```

Comment:
what about simplifying it?
```suggestion
        policy_events = [LoopInterrupted(True)] if entered_unhappy_path else []
```